app/src/main/java/de/vectordata/skynet/ui/base/SkynetActivity.java

```diff
@@ -46,6 +53,19 @@ public void onAuthenticationFailed(AuthenticationFailedEvent event) {
             Dialogs.showMessageBox(this, R.string.error_header_connect, R.string.error_invalid_session);
     }
 
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSyncFinished(SyncFinishedEvent event) {
+        int corruptedMessages = SkynetContext.getCurrent().getNetworkManager().getLastSyncCorruptedMessages();
+        if (corruptedMessages > 0) {
```

Comment:
What if you receive a corrupted message after sync?

Reply:
The idea here was to not show a dialog every time a corrupted message is received, as to not annoy the user (e.g. if someone does this with a malicious client, you could make anyone's app basically unusable)

Reply:
As mentioned above, messages from others can be handled with `MessageState.Corrupted`. Corrupted messages with `MessageFlags.Loopback` can annoy the user. If you want to use a malicious client, you will have to live with the consequences.

Creating only one dialog for the whole sync process is a good idea. You could implement something like this for loopback messages in `PacketHandler.java`:

```java
if (inSync) {
    showDialogForCorruptedMessage();
} else {
    numCorruptedMessages++;
}
```

Reply:
But that means that every single corrupted message will show a dialog. What I meant with malicious client was the following scenario:

1. Open a direct channel with someone
2. Start spamming corrupted messages
3. The target will be spammed with dialog messages the second he opens Skynet. He cannot block the attacker because there are dialogs in the way. The account is basically unusable.

As you said only chat messages can be corrupted (as of now, If we have something like the discussed BeginCall, we would have some other issues). I would only show the dialog after sync and keep everything else to a per-message info.

Reply:
Only corrupted messages with `MessageFlags.Loopback` will show a dialog (see my suggestion in `PacketHandler.java:105`)

Please implement this second dialog for a single corrupted message that was received after completing the sync like described above.